cli/src/cli_util.rs

```diff
@@ -2824,19 +2824,19 @@ Discard the conflicting changes with `jj restore --from {}`.",
 }
 
 /// Prints warning about explicit paths that don't match any of the tree
-/// entries.
-pub fn print_unmatched_explicit_paths<'a>(
+/// entries, returning the unmatched paths.
+pub fn print_unmatched_explicit_paths<'a, 'expr>(
     ui: &Ui,
     workspace_command: &WorkspaceCommandHelper,
-    expression: &FilesetExpression,
+    expression: &'expr FilesetExpression,
     trees: impl IntoIterator<Item = &'a MergedTree>,
-) -> io::Result<()> {
+) -> io::Result<Vec<&'expr RepoPath>> {
```

Comment:
Can you revert this change? It's odd that `print()` function returns a list of paths to be printed, and we can easily test the same condition by caller now.

Reply:
Good point- not necessary anymore.